packages/camera/camera/android/src/main/java/io/flutter/plugins/camera/features/exposurepoint/ExposurePoint.java

```
@@ -0,0 +1,82 @@
+package io.flutter.plugins.camera.features.exposurepoint;
+
+import android.hardware.camera2.CaptureRequest;
+import android.hardware.camera2.params.MeteringRectangle;
+import android.util.Log;
+import io.flutter.plugins.camera.CameraProperties;
+import io.flutter.plugins.camera.features.CameraFeature;
+import io.flutter.plugins.camera.features.Point;
+import io.flutter.plugins.camera.features.regionboundaries.CameraRegions;
+import java.util.concurrent.Callable;
+
+/** Exposure point controls where in the frame exposure metering will come from. */
+public class ExposurePoint implements CameraFeature<Point> {
+  // Used later to always get the correct camera regions instance.
+  private final Callable<CameraRegions> getCameraRegions;
+  private boolean isSupported;
+  private Point currentSetting = new Point(0.0, 0.0);
+
+  public ExposurePoint(
+      CameraProperties cameraProperties, Callable<CameraRegions> getCameraRegions) {
+    this.getCameraRegions = getCameraRegions;
+    this.isSupported = checkIsSupported(cameraProperties);
+  }
+
+  @Override
+  public String getDebugName() {
+    return "ExposurePoint";
+  }
+
+  @Override
+  public Point getValue() {
+    return currentSetting;
+  }
+
+  @Override
+  public void setValue(Point value) {
+    this.currentSetting = value;
+
+    try {
+      if (value.x == null || value.y == null) {
+        getCameraRegions.call().resetAutoExposureMeteringRectangle();
+      } else {
+        getCameraRegions.call().setAutoExposureMeteringRectangleFromPoint(value.x, value.y);
+      }
+    } catch (Exception e) {
+      e.printStackTrace();
```

@accoutts While writing unit tests I encountered this line and was wondering if this would be the correct way to handle the exception (i.e. I assume this will just suppress the exception and continue executing the flow but I am not sure). Can you confirm?

---

Ah great catch. In general it seems like the right move is to use the dartMessenger to send an error event, unless there's a dart `Result` object available where you would then just call `result.error(...)`